src/Illuminate/Database/Query/Builder.php

```diff
@@ -1427,31 +1441,33 @@ public function simplePaginate($perPage = 15, $columns = ['*'], $pageName = 'pag
      */
     public function getCountForPagination($columns = ['*'])
     {
-        $this->backupFieldsForCount();
+        $selectSql = $this->toSql();
```

Comment:
This line can surely be moved inside the if block so it's not executed if we don't need it?